services/cd-service/pkg/repository/repository.go

```diff
@@ -2599,7 +2628,7 @@ func (s *State) UpdateEnvironmentInternal(ctx context.Context, transaction *sql.
 		envInGroup.Locks = env.Locks
 	}
 
-	if apps, err := s.GetEnvironmentApplications(ctx, transaction, envName); err != nil {
+	if apps, err := s.GetApplications(ctx, transaction); err != nil {
```

Comment:
Why do we get all applications here instead of just the environment apps?

Reply:
Because Environment applications are only the applications that have a release on that environment. There can be a lock on an application that doesn't have any releases on that environment.